src/crm/dynamicsWebApi.ts

```diff
@@ -59,26 +61,63 @@ const onRejected = (error: AxiosError) => {
   return Promise.reject(error);
 };
 
-const getTestStationEntities = async (requestUrl: string): Promise<DynamoTestStation[]> => {
+const getAccountsEntities = async (requestUrl: string): Promise<DynamicsTestStation[]> => {
   const siteList = (await getSecret(config.crm.siteList)).split(',');
+
+  interface AccountsFormat {
+    value: DynamicsTestStation[];
+  }
+  return lastValueFrom(
+    axios.get<AccountsFormat>(requestUrl).pipe(
+      map((data) => data.data),
+      map((data: AccountsFormat) => data.value.filter((obj) => siteList.includes(obj.dvsa_premisecodes))),
+    ),
+  );
+};
+
+const addConnectionsEmails = async (stationEntries: DynamicsTestStation[]): Promise<DynamoTestStation[]> => {
+  interface ConnectionsFormat {
+    value: DynamicsConnections[]
+  }
+
+  const ceUrl = config.crm.ceBaseUrl;
+  const { ceRoleId } = config.crm;
+
+  const testStationEvents = stationEntries.map(async (station) => {
+    const requestUrl = `${ceUrl}/connections?$select=_record2id_value&$expand=record1id_account($select=accountid,accountnumber),record2id_contact($select=emailaddress1)&$filter=_record1id_value%20eq%20${station.accountid}%20and%20_record2roleid_value%20eq%${ceRoleId}%20and%20statuscode%20eq%201`;
+
+    const testStationEvent = createDynamoTestStation(station);
+    await lastValueFrom(
+      axios.get<ConnectionsFormat>(requestUrl).pipe(
+        map((data) => data.data),
+        map((data: ConnectionsFormat) => data.value.forEach((connection) => {
+          testStationEvent.testStationEmails.push(connection.record2id_contact.emailaddress1);
```

Comment:
I think here instead of mapping you can assign one array to the other ?

Reply:
That's what mapping essentially does. But in this case there's an asynchronous call for each station entry in the array which map allows you to run in parallel, which is why there's a `Promise.all` at the end.